streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -1047,9 +1048,9 @@ private static Metrics getMetrics(final StreamsConfig config, final Time time, f
                 // and then resize them later
                 streamThread = createAndAddStreamThread(0L, 0L, threadIdx);
                 final int numLiveThreads = getNumLiveStreamThreads();
-                resizeThreadCacheAndBufferMemory(numLiveThreads + 1);
+                resizeThreadCacheAndBufferMemory(numLiveThreads);
                 log.info("Adding StreamThread-{}, there are now {} threads with cache size/max buffer size values as {} per thread.",
-                        threadIdx, numLiveThreads + 1, getThreadCacheAndBufferMemoryString());
+                        threadIdx, numLiveThreads, getThreadCacheAndBufferMemoryString());
```

Comment:
I think we should add tests for it, to make sure after adding a thread, the cache size and buffer memory is set as what we expected.

Reply:
@showuon , there are some tests for this here like : https://github.com/vamossagar12/kafka/blob/84c75ce44167e5e1bf8748a7ff9ba95309cd3d7e/streams/src/test/java/org/apache/kafka/streams/integration/AdjustStreamThreadCountTest.java#L373. which were added as part of original PR. Do you think they are enough?

Reply:
ok.. yeah there are 2 more tests for threadReplacement which seem to be printing out an extra thread count. But I agree, a test for thread addition would be good..